src/System.IO.Packaging/src/System/IO/Packaging/ContentType.cs

```diff
@@ -0,0 +1,727 @@
+        {
+            get
+            {
+                EnsureParameterDictionary();
+                return _parameterDictionary.GetEnumerator();
+            }
+        }
+        #endregion Internal Properties
+
+        //------------------------------------------------------
+        //
+        //  Internal Methods
+        //
+        //------------------------------------------------------
+
+        #region Internal Methods
+
+        /// <summary>
+        /// This method does a strong comparison of the content types, as parameters are not allowed. 
+        /// We only compare the type and subType values in an ASCII case-insensitive manner.
+        /// Parameters are not allowed to be present on any of the content type operands.
+        /// </summary>
+        /// <param name="contentType">Content type to be compared with</param>
+        /// <returns></returns>
+        internal bool AreTypeAndSubTypeEqual(ContentType contentType)
+        {
+            return AreTypeAndSubTypeEqual(contentType, false);
+        }
+
+        /// <summary>
+        /// This method does a weak comparison of the content types. We only compare the
+        /// type and subType values in an ASCII case-insensitive manner.
+        /// Parameter and value pairs are not used for the comparison. 
+        /// If you wish to compare the paramters too, then you must get the ParameterValuePairs from
+        /// both the ContentType objects and compare each parameter entry.
+        /// The allowParameterValuePairs parameter is used to indicate whether the 
+        /// comparison is tolerant to parameters being present or no.
+        /// </summary>
+        /// <param name="contentType">Content type to be compared with</param>
+        /// <param name="allowParameterValuePairs">If true, allows the presence of parameter value pairs. 
+        /// If false, parameter/value pairs cannot be present in the content type string.
+        /// In either case, the parameter value pair is not used for the comparison.</param>
+        /// <returns></returns>
+        internal bool AreTypeAndSubTypeEqual(ContentType contentType, bool allowParameterValuePairs)
+        {
+            bool result = false;
+
+            if (contentType != null)
+            {
+                if (!allowParameterValuePairs)
+                {
+                    //Return false if this content type object has parameters
+                    if (_parameterDictionary != null)
+                    {
+                        if (_parameterDictionary.Count > 0)
+                            return false;
+                    }
+
+                    //Return false if the content type object passed in has parameters
+                    Dictionary<string, string>.Enumerator contentTypeEnumerator;
+                    contentTypeEnumerator = contentType.ParameterValuePairs;
+                    contentTypeEnumerator.MoveNext();
```

Comment:
What was the verdict on this?  Does there need to be a check on the return value of MoveNext?